test/ProductConstructionService.WorkItem.Tests/WorkItemScopeTests.cs

```diff
@@ -167,17 +193,25 @@ public async Task MultipleProcessorsWithoutFactoryMethodTest()
 
         IServiceProvider serviceProvider = _services.BuildServiceProvider();
 
-        WorkItemScopeManager scopeManager = new(false, serviceProvider, Mock.Of<ILogger<WorkItemScopeManager>>());
+        Mock<IRedisCacheFactory> cacheFactory = new();
+        cacheFactory.Setup(f => f.Create(It.IsAny<string>())).Returns(new FakeRedisCache());
+
+        WorkItemProcessorState state = new(
```

Comment:
`WorkItemProcessorState` is created the same way 4 times in this class but it seems we aren't using it directly, just to construct `WorkItemScopeManager` later on. Maybe it can be a private method or even an attribute and be reused whenever its needed?

Reply:
done